pkgs/applications/networking/onedriver/default.nix

```diff
@@ -0,0 +1,56 @@
+{ buildGoModule
+, fetchFromGitHub
+, lib
+, pkg-config
+, webkitgtk
+, glib
+, fuse
+, installShellFiles
+}:
+
+buildGoModule rec {
+  pname = "onedriver";
+  version = "0.13.0-2";
+
+  src = fetchFromGitHub {
+      owner = "jstaf";
+      repo = "onedriver";
+      rev = "v${version}";
+      hash = "sha256-Bcjgmx9a4pTRhkzR3tbOB6InjvuH71qomv4t+nRNc+w=";
+  };
```

Comment:
suggestion:

```suggestion
let 
  pname = "onedriver";
  version = "0.13.0-2";
  
  src = fetchFromGitHub {
      owner = "jstaf";
      repo = "${pname}";
      rev = "v${version}";
      hash = "sha256-Bcjgmx9a4pTRhkzR3tbOB6InjvuH71qomv4t+nRNc+w=";
  };
in buildGoModule {
 inherit pname version src;
```

Reply:
I did the change as you suggested, but since I'm very new with nix, could you please tell me the reason why it's better to have those variables outside of the derivation? I'm thinking it may be to avoid them being overridable but I'm not sure..

Reply:
Sure 😃

`rec` isn't "bad", but it can lead to infinite recursion, which isn't possible with `let ... in` bindings.

Personally, I use `rec` when I just need to refer to one attribute in the same set, but if it's more, I find `let` bindings more readable.